### PR TITLE
feat(argo-cd): Support AWS ALB Ingress with gRPC

### PR DESCRIPTION
**Pull request by toneill818**
Signed-off-by: Thomas O'Neill <toneill818@gmail.com>

Fixes Issue #803 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/argocd-server/alb-grpc-service.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{{- if .Values.server.ingressGrpc.enabled -}}
+{{- if .Values.server.ingressGrpc.isAWSALB -}}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-server/alb-grpc-service.yaml:2]: Is there a reason, why you didn't write it like this?
```suggestion
{{- if and .Values.server.ingressGrpc.enabled .Values.server.ingressGrpc.isAWSALB -}}
```

Comment (toneill818) [reply charts/argo-cd/templates/argocd-server/alb-grpc-service.yaml:2]: Nope, no reason, I can change it to the one line if statement.

Comment (toneill818) [reply charts/argo-cd/templates/argocd-server/alb-grpc-service.yaml]: Changed the if statements.

File: charts/argo-cd/templates/argocd-server/alb-grpc-service.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    alb.ingress.kubernetes.io/backend-protocol-version: HTTP2 #This tells AWS to send traffic from the ALB using HTTP2. Can use GRPC as well if you want to leverage GRPC specific features
+  labels:
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" .Values.server.name) | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (toneill818) [reply charts/argo-cd/templates/argocd-server/alb-grpc-service.yaml]: added the `-grpc` suffix for the labels

**Review by mbevc1 — APPROVED** [review]

**Pull request merged**